Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,298 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
+#import <VAListHelper.h>
 
-#include "Foundation/NSOrderedSet.h"
+// Needed to make sure that the returned array behaves like a one way proxy to an array
+@implementation _NSProxyOrderedSetArray
+- (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet {
+    if (self = [super init]) {
+        _array = orderedSet->_arrayContainer;
+    }
+
+    return self;
+}
+
+#pragma mark NSArray
```

Comment:
I added it for code organization, given the classes don't really use them anyway for grouping, i removed it.